torch/_dynamo/guards.py

```diff
@@ -342,11 +343,19 @@ def DICT_KEYS(self, guard):
         code.append(f"___check_type_id({ref}, {self.id_ref(t)})")
         param_key_ids = set(dict_param_key_ids(value))
         const_keys = set(dict_const_keys(value))
+        if any(isinstance(k, enum.Enum) for k in const_keys):
+            # Workaround repr(Enum) returns invalid global reference before python 3.11
+            # https://peps.python.org/pep-0663/
+            const_keys_str = f"{set([str(k) if isinstance(k, enum.Enum) else repr(k) for k in const_keys])}".replace(
```

Comment:
Python sets don't preserve order, we should force this to be deterministic.

Reply:
Oh, here we generate code to check the two sets are the same, so we don't need to preserve order? The generated code looks like:
```
set(z.keys()) == {"abc", MyEnum.FOO, MyEnum.BAR, 5}
```